x-pack/legacy/plugins/infra/server/lib/adapters/log_entries/kibana_log_entries_adapter.ts

```diff
@@ -319,6 +374,34 @@ function getLookupIntervals(start: number, direction: 'asc' | 'desc'): Array<[nu
   return intervals;
 }
 
+function mapHitsToLogEntryDocuments(
+  hits: LogItemHit[],
+  timestampField: string,
+  fields: string[]
+): LogEntryDocument[] {
+  return hits.map(hit => {
+    const logFields = fields.reduce<{ [fieldName: string]: JsonValue }>(
+      (flattenedFields, field) => {
+        if (has(field, hit._source)) {
```

Comment:
I hear you, but I don't want to dedicate much time to this function. It will go away in a separate PR.

Right now the API is handled in three files

- `lib/adapters/kibana_log_entries_adapter`, which connects with Elasticsearch.
- `lib/domains/log_entries_domain`, which connects the adapter with the route files
- The route file.

This function in the adapter takes the ES response and transforms it onto a `LogEntryDocument`, an intermediate format for the `domain` that then gets transformed again in the route.

I had a chat with @weltenwort and @Kerry350 a couple of weeks ago about how this code was organised, and the conclusion was to merge the `domain` and the `adapter` files into one. Once we do that we don't need an intermediate format anymore and this function will go away.

---

I will take your comment into account when I join the two files in one. I agree with you that it's not clear straight away what is in `_source`. I guess there's some documentation somewhere of how filebeat stores the log metadata in ES. We could just add a comment with a link to it.